mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/index/impl/MongoDBIndexSpec.java

```diff
@@ -193,4 +194,38 @@ public String toString() {
 		return sb.toString();
 	}
 
+	@Override
+	public boolean equals(Object o) {
+		if ( this == o ) {
+			return true;
+		}
+		if ( o == null || getClass() != o.getClass() ) {
+			return false;
+		}
+		MongoDBIndexSpec that = (MongoDBIndexSpec) o;
+		return Objects.equals( collection, that.collection ) &&
+				Objects.equals( indexKeys, that.indexKeys );
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash( collection, indexKeys );
+	}
+
+	/**
+	 * this method could be used to choose between different configuration
+	 */
+	@Override
+	public int compareTo(MongoDBIndexSpec o) {
+		// choose the one with unique over a non unique
+		if ( this.getOptions().isUnique() && !o.getOptions().isUnique() ) {
```

Comment:
I don't think it's a good idea to use the compareTo in this scenario. I think it is much easier to understand if you create a method in the class where you need this kind of comparison.

Reply:
I agree